src/ConfigCatContext.tsx

```diff
@@ -14,4 +14,23 @@ const ConfigCatContext = React.createContext<ConfigCatContextData | undefined>(
 
 ConfigCatContext.displayName = "ConfigCatContext";
 
+const ConfigCatContextRegistry = new Map<string, React.Context<ConfigCatContextData | undefined>>();
+
+export function getOrCreateConfigCatContext(configId: string): React.Context<ConfigCatContextData | undefined> {
+
+  let context: React.Context<ConfigCatContextData | undefined> | undefined = ConfigCatContextRegistry.get(configId.toLocaleLowerCase());
+
+  if (!context) {
+    context = React.createContext<ConfigCatContextData | undefined>(
+      void 0
+    );
+
+    context.displayName = "ConfigCatContext_" + configId;
+
+    ConfigCatContextRegistry.set(configId.toLocaleLowerCase(), context);
```

Comment:
Using locale-sensitive lower case for this purpose may lead to obscure bugs, so I advise against it. (See also: https://stackoverflow.com/a/34283467/8656352)

```suggestion
  let context: React.Context<ConfigCatContextData | undefined> | undefined = ConfigCatContextRegistry.get(configId.toLowerCase());

  if (!context) {
    context = React.createContext<ConfigCatContextData | undefined>(
      void 0
    );

    context.displayName = "ConfigCatContext_" + configId;

    ConfigCatContextRegistry.set(configId.toLowerCase(), context);
```

Reply:
wow :)
I was changed the method to the proper one.